src/components/Questionnaire/QuestionTypes/MedicationRequestQuestion.tsx

```diff
@@ -74,6 +78,94 @@ interface MedicationRequestQuestionProps {
   ) => void;
   disabled?: boolean;
   encounterId: string;
+  errors?: QuestionValidationError[];
+}
+
+const MEDICATION_REQUEST_FIELDS = {
+  DOSAGE: {
+    key: "dosage_instruction.dose",
+    required: true,
+    validate: (value: MedicationRequest["dosage_instruction"][0]) => {
+      // Check if either dose_quantity or dose_range exists
+      return !!(
+        value?.dose_and_rate?.dose_quantity || value?.dose_and_rate?.dose_range
+      );
+    },
+  },
+  FREQUENCY: {
+    key: "dosage_instruction.frequency",
+    required: true,
+    validate: (value: MedicationRequest["dosage_instruction"][0]) => {
+      // Check if either timing exists or as_needed_boolean is true
+      return !!(value?.timing || value?.as_needed_boolean);
+    },
+  },
+  DURATION: {
+    key: "dosage_instruction.duration",
+    required: false,
+    validate: (value: MedicationRequest["dosage_instruction"][0]) => {
+      // If timing exists, bounds_duration must be present with value and unit
+      if (value?.timing) {
+        const duration = value.timing.repeat.bounds_duration;
+        return !!(duration?.value && duration?.unit);
+      }
+      return true; // No validation needed if no timing
+    },
+  },
+} as const;
+
+export function validateMedicationRequestQuestion(
+  values: MedicationRequest[],
+  questionId: string,
+  t: (key: string) => string,
+): QuestionValidationError[] {
+  return values.reduce((errors: QuestionValidationError[], value, index) => {
+    // Skip validation for medications marked as entered_in_error
+    if (value.status === "entered_in_error") return errors;
+
+    // Validate each dosage instruction
+    const dosageInstruction = value.dosage_instruction[0];
+    if (!dosageInstruction) {
+      return [
+        ...errors,
+        {
+          question_id: questionId,
+          error: t("field_required"),
+          type: "validation_error",
+          field_key: "dosage_instruction",
+          index,
+        },
+      ];
+    }
+
+    // Validate using the fields
+    const fieldErrors = validateFields(
+      {
+        [MEDICATION_REQUEST_FIELDS.DOSAGE.key]: dosageInstruction,
+        [MEDICATION_REQUEST_FIELDS.FREQUENCY.key]: dosageInstruction,
+        [MEDICATION_REQUEST_FIELDS.DURATION.key]: dosageInstruction,
+      },
+      questionId,
+      MEDICATION_REQUEST_FIELDS,
+      index,
+    );
+
+    // Map error messages to be more specific
+    return [
+      ...errors,
+      ...fieldErrors.map((error) => ({
+        ...error,
+        error:
+          error.field_key === MEDICATION_REQUEST_FIELDS.DOSAGE.key
+            ? t("field_required")
+            : error.field_key === MEDICATION_REQUEST_FIELDS.FREQUENCY.key
+              ? t("field_required")
+              : error.field_key === MEDICATION_REQUEST_FIELDS.DURATION.key
+                ? t("field_required")
+                : error.error,
```

Comment:
this can be simplified a lot

```suggestion
          (["DOSAGE", "FREQUENCY", "DURATION"] as const)
            .some((attr) => MEDICATION_REQUEST_FIELDS[attr].key === error.field_key) 
              ? t("field_required") 
              : error.error,
```